controllers/scopeinstance_controller.go

```diff
@@ -58,47 +67,166 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
-	// get 	the scope template
+	log.Log.Info("After getting ScopeInstance", "namespace", req.NamespacedName)
+
 	st := &operatorsv1.ScopeTemplate{}
 	if err := r.Client.Get(ctx, client.ObjectKey{Name: in.Spec.ScopeTemplateName}, st); err != nil {
+		// 4. Updating a ScopeInstance CR to reference a non-existent ScopeTemplate causes existing RBAC to be deleted.
+		// You can check here if RBAC present if yes then delete it
+		// delete RBAC and ScopeInstance both
+		log.Log.Info("throwing error check 4th condition scope template")
 		return ctrl.Result{}, err
 	}
 
 	log.Log.Info("ScopeTemplate found", "name", st.Name)
 
+	existingRoleBindingList := &rbacv1.RoleBindingList{}
+	existingClusterRoleBindingList := &rbacv1.ClusterRoleBindingList{}
+
+	// Get the existing ClusterRoleBinding
+	if err := r.Client.List(ctx, existingClusterRoleBindingList, client.MatchingLabels{
+		"labels": "scopeInstanceController",
+	}); err != nil {
+		log.Log.Info("Error getting object", "error", err)
+	}
+
+	// look at namespaces - map of run time objects - think of this later
+	// Get the existing RoleBindings
+	if err := r.Client.List(ctx, existingRoleBindingList, client.MatchingLabels{
+		"labels": "scopeInstanceController",
+	}); err != nil {
+		log.Log.Info("Error getting object", "error", err)
```

Comment:
We cannot risk creating duplicate roleBindings, return an error if the list fails.
```suggestion
		return ctrl.Result{}, err
```

Reply:
Updated.